src/ui/public/directives/paginate.js

```diff
@@ -40,6 +40,12 @@ uiModules.get('kibana')
     },
     controllerAs: 'paginate',
     controller: function ($scope, $document) {
+      // Unfortunately this controller relies on inherited scope. Lets not fail if onPageChanged happens to not be
+      // implemented.
+      if (!$scope.onPageChanged) {
```

Comment:
++ No reason, and since you gave a reason for the alternate approach, I'll switch it over!